### PR TITLE
[DOCS] Updating Weight Compression Article

### PR DESCRIPTION
**Pull request by sgolebiewski-intel**
Rearranging information in the `LLM Weight Compression` article.

### CODE REVIEW

File: docs/articles_en/openvino-workflow/model-optimization-guide/weight-compression.rst [hunk 1]
```diff
@@ -6,38 +6,38 @@ LLM Weight Compression
...
 
 
-Weight compression is a technique for enhancing the efficiency of models,
-especially those with large memory requirements. This method reduces the model's
+Weight compression enhances the efficiency of models,
+which have large memory requirements. This method reduces the model's
 memory footprint, a crucial factor for Large Language Models (LLMs).
 
 Unlike full model quantization, where weights and activations are quantized,
 weight compression in `Neural Network Compression Framework (NNCF) <https://github.com/openvinotoolkit/nncf>`__
```
Comment (kblaszczak-intel) [inline docs/articles_en/openvino-workflow/model-optimization-guide/weight-compression.rst:17]:
```suggestion
Unlike full model quantization, where both weights and activations are quantized, it
```

File: docs/articles_en/openvino-workflow/model-optimization-guide/weight-compression.rst [hunk 2]
```diff
@@ -6,38 +6,38 @@ LLM Weight Compression
...
-with larger models.
+.. note::
+
+   With smaller language models (i.e. less than 1B parameters), weight
+   compression may result in more accuracy reduction than with larger models.
+   Therefore, weight compression is recommended for use with LLMs only.
 
 LLMs and other models that require
 extensive memory to store the weights during inference can benefit
 from weight compression as it:
```
Comment (kblaszczak-intel) [inline docs/articles_en/openvino-workflow/model-optimization-guide/weight-compression.rst:32]: Maybe we don't need this piece. It just restates stuff. But we need to check what "other models" means in this case.

File: docs/articles_en/openvino-workflow/model-optimization-guide/weight-compression.rst [hunk 3]
```diff
@@ -46,17 +46,15 @@ provides weight quantization to 8 and 4-bit integer data types as a compression
...
-Compress Model Weights
-######################
+Compression Methods (8-bit vs. 4-bit)
+#####################################
 
 **8-bit weight quantization** method offers a balance between model size reduction and
 maintaining accuracy, which usually leads to significant performance improvements for
 Transformer-based models. Models with 8-bit compressed weights are performant on the
 vast majority of supported CPU and GPU platforms. By default, weights are compressed
 asymmetrically to "INT8_ASYM" mode.
```
Comment (kblaszczak-intel) [inline docs/articles_en/openvino-workflow/model-optimization-guide/weight-compression.rst:56]:
```suggestion
**8-bit weight quantization** offers a good balance between reducing the size and lowering the
accuracy of a model. It usually results in significant improvements for transformer-based models
and guarantees good model performance for a vast majority of supported CPU and GPU platforms.
By default, weights are compressed asymmetrically to "INT8_ASYM" mode.
```

File: docs/articles_en/openvino-workflow/model-optimization-guide/weight-compression.rst [hunk 4]
```diff
@@ -46,17 +46,15 @@ provides weight quantization to 8 and 4-bit integer data types as a compression
...
 
-
 The code snippet below shows how to do asymmetrical 8-bit quantization of the model weights
 represented in OpenVINO IR using NNCF:
```
Comment (kblaszczak-intel) [inline docs/articles_en/openvino-workflow/model-optimization-guide/weight-compression.rst:59]:
```suggestion
Here is an example of code using NNCF to perform asymmetrical 8-bit weight quantization of
a model in the OpenVINO IR format:
```

File: docs/articles_en/openvino-workflow/model-optimization-guide/weight-compression.rst [hunk 5]
```diff
@@ -70,13 +68,77 @@ represented in OpenVINO IR using NNCF:
...
+         model = OVModelForCausalLM.from_pretrained(model_id, export=True, load_in_8bit=True)
+
+         # Inference
+         tokenizer = AutoTokenizer.from_pretrained(model_id)
+         pipe = pipeline("text-generation", model=model, tokenizer=tokenizer)
+         phrase = "The weather is"
+         results = pipe(phrase)
+         print(results)
+
+The model can be also :ref:`saved into a compressed format <save_pretrained>`,
```
Comment (kblaszczak-intel) [inline docs/articles_en/openvino-workflow/model-optimization-guide/weight-compression.rst:134]:
```suggestion
The model can also be :ref:`saved into a compressed format <save_pretrained>`,
```

File: docs/articles_en/openvino-workflow/model-optimization-guide/weight-compression.rst [hunk 6]
```diff
@@ -70,13 +68,77 @@ represented in OpenVINO IR using NNCF:
...
+resulting in a smaller binary file.
+
 
 **4-bit weight quantization** method stands for an INT4-INT8 mixed-precision weight quantization,
 where INT4 is considered as the primary precision and asymmetric INT8 is the backup one.
 It usually results in a smaller model size and lower inference latency, although the accuracy
```
Comment (kblaszczak-intel) [inline docs/articles_en/openvino-workflow/model-optimization-guide/weight-compression.rst:140]:
```suggestion
**4-bit weight quantization** is actually a mixed-precision compression, primarily INT4 and
a backup asymmetric INT8 precisions. It produces a smaller model, offering lower inference latency but potentially noticeable accuracy degradation, depending on the model.
```

File: docs/articles_en/openvino-workflow/model-optimization-guide/weight-compression.rst [hunk 7]
```diff
@@ -70,13 +68,77 @@ represented in OpenVINO IR using NNCF:
...
-degradation could be higher, depending on the model.
+degradation can be higher, depending on the model.
```
Comment (kblaszczak-intel) [inline docs/articles_en/openvino-workflow/model-optimization-guide/weight-compression.rst:141]:
```suggestion

```

File: docs/articles_en/openvino-workflow/model-optimization-guide/weight-compression.rst (hunk 2 above)
Comment (sgolebiewski-intel) [reply docs/articles_en/openvino-workflow/model-optimization-guide/weight-compression.rst]: > Maybe we don't need this piece. It just restates stuff. But we need to check what "other models" means in this case.

I have changed it with the [commit](https://github.com/openvinotoolkit/openvino/pull/27432/commits/b354ec794b5d3ef1ef25ae26a61cddfa9ce6dff5)
as agreed with @AlexKoff88

**Comment by AlexKoff88** [comment]
A few main comments from my side:
- I intentionally removed the first table with two tabs of code because one of them is an example of optimization with Optimum-Intel. But actually, there is a dedicated section about that.
- Let also add an `optimum-cli` example in after the API usage
- I would like to see a cross reference and information that the optimized models can be run with OpenVINO GenAI

File: docs/articles_en/openvino-workflow/model-optimization-guide/weight-compression.rst [hunk 8]
```diff
@@ -70,13 +65,100 @@ represented in OpenVINO IR using NNCF:
...
+         from transformers import AutoTokenizer, pipeline
+
+         # Load and compress a model from Hugging Face.
+         model_id = "microsoft/Phi-3.5-mini-instruct"
+         model = OVModelForCausalLM.from_pretrained(
+             model_id,
+             export=True,
+             quantization_config=OVWeightQuantizationConfig(
+                 bits=8,
+                 quant_method="awq",
```
Comment (AlexKoff88) [inline docs/articles_en/openvino-workflow/model-optimization-guide/weight-compression.rst:131]: This is a config for int4 quantization (not int8) and it should be moved below with the change of `bits=4` and the comment above.
Also you have the command-line variant of it which is for int4 as you can see `optimum-cli export openvino --model microsoft/Phi-3.5-mini-instruct --weight-format int4 --awq --scale-estimation --dataset wikitext2 --group-size 64 --ratio 1.0 ov_phi-3.5-mini-instruct`.

For INT8 it looks like:
```
quantization_config=OVWeightQuantizationConfig(bits=8)
```
and
```
optimum-cli export openvino --model microsoft/Phi-3.5-mini-instruct --weight-format int8 ov_phi-3.5-mini-instruct
```

Comment (sgolebiewski-intel) [reply docs/articles_en/openvino-workflow/model-optimization-guide/weight-compression.rst]: I have applied necessary changes in: https://github.com/openvinotoolkit/openvino/pull/27432/commits/5b49f023635c0ac453d5ead14e57f4fc149407c3

**Review by AlexKoff88 — APPROVED** [review]

**Pull request merged**